apps/io-services-cms-webapp/src/publicator/__tests__/request-publication-handler.test.ts

```diff
@@ -129,9 +133,32 @@ describe("Service Publication Handler", () => {
     await handleQueueItem(
       context,
       autoPublishQueueItem,
-      mockFsmLifecycleClient
+      mockFsmPublicationClient
     )();
-    expect(mockFsmLifecycleClient.unpublish).toBeCalledTimes(1);
-    expect(mockFsmLifecycleClient.unpublish).toBeCalledWith(aService.id);
+    expect(mockFsmPublicationClient.unpublish).toBeCalledTimes(1);
+    expect(mockFsmPublicationClient.unpublish).toBeCalledWith(aService.id);
+  });
+
+  it("handleQueueItem should be void 0 if unpublish return an FsmItemNotFoundError", async () => {
+    const QueueItem = {
+      id: aService.id,
+      kind: "RequestUnpublicationItem",
+    } as unknown as Json;
+    const context = createContext();
+
+    class FSMError extends Error {
+      public kind = "FsmItemNotFoundError";
+    }
+    const mockFsmPublicationClient = {
+      unpublish: vi.fn(() => TE.right(new FSMError())),
```

Comment:
Shouldn't it be TE.left in case of an error when applying a transition?

Reply:
i get i serialized error if i use TE.left and i don't know why...

Reply:
in order to solve the issue you can:
- Remove the FSMError class
- Update mockFsmPublicationClient in the following way:
``` ts
    const mockFsmPublicationClient = {
      unpublish: vi.fn(() => TE.left(new FsmItemNotFoundError(aService.id))),
    } as unknown as ServicePublication.FsmClient;
```